### PR TITLE
Rework of settings panel

### PR DESCRIPTION
**Pull request by keerath**
rework-of-settings-panel

### CODE REVIEW

**Comment by ScrapCodes** [comment]
null
java.lang.NullPointerException
	at com.imaginea.kodebeagle.ui.SettingsPanel.reset(SettingsPanel.java:420)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel$1.compute(ConfigurableCardPanel.java:56)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel$1.compute(ConfigurableCardPanel.java:51)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:884)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.create(ConfigurableCardPanel.java:51)
	at com.intellij.openapi.options.newEditor.ConfigurableEditor$1.create(ConfigurableEditor.java:64)
	at com.intellij.openapi.options.newEditor.ConfigurableEditor$1.create(ConfigurableEditor.java:61)
	at com.intellij.ui.CardLayoutPanel.createValue(CardLayoutPanel.java:73)
	at com.intellij.ui.CardLayoutPanel.select(CardLayoutPanel.java:101)
	at com.intellij.ui.CardLayoutPanel.access$000(CardLayoutPanel.java:35)
	at com.intellij.ui.CardLayoutPanel$1$1.run(CardLayoutPanel.java:118)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:745)
	at java.awt.EventQueue.access$300(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:706)
	at java.awt.EventQueue$3.run(EventQueue.java:704)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:715)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:154)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:182)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:221)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:219)
	at java.awt.Dialog.show(Dialog.java:1082)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:778)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:465)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1614)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1571)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:143)
	at com.imaginea.kodebeagle.action.EditSettingsAction.actionPerformed(EditSettingsAction.java:37)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:182)
	at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:162)
	at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:126)
	at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:303)
	at java.awt.Component.processEvent(Component.java:6281)
	at java.awt.Container.processEvent(Container.java:2229)
	at java.awt.Component.dispatchEventImpl(Component.java:4872)
	at java.awt.Container.dispatchEventImpl(Container.java:2287)
	at java.awt.Component.dispatchEvent(Component.java:4698)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4832)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4492)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4422)
	at java.awt.Container.dispatchEventImpl(Container.java:2273)
	at java.awt.Window.dispatchEventImpl(Window.java:2719)
	at java.awt.Component.dispatchEvent(Component.java:4698)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:747)
	at java.awt.EventQueue.access$300(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:706)
	at java.awt.EventQueue$3.run(EventQueue.java:704)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.awt.EventQueue$4.run(EventQueue.java:718)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:717)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:565)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)

**Comment by keerath** [comment]
Fixed NP Exception

**Comment by ScrapCodes** [comment]
Let's increase the lowest for `Result Size` and `Top Results Count` to one.

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/ui/PatternFilterEditor.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
+/*
```
Comment (keerath) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/ui/PatternFilterEditor.java]: File referred from : https://github.com/JetBrains/intellij-community/blob/417010e6540243ff7943072c4ef17ed174d05b6d/java/openapi/src/com/intellij/ui/classFilter/ClassFilterEditor.java

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/ui/PatternFilterEditorAddDialog.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
```
Comment (keerath) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/ui/PatternFilterEditorAddDialog.java]: Referred From https://github.com/JetBrains/intellij-community/blob/7c8377934c3c433b75ef79c7bc28dddb8cc4b21c/java/openapi/src/com/intellij/ui/classFilter/ClassFilterEditorAddDialog.java

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/model/Settings.java [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+            }
+            Limits limits = (Limits) obj;
+            return this.getLinesFromCursor() == limits.getLinesFromCursor()
+                    && this.getResultSize() == limits.getResultSize()
+                    && this.getTopCount() == limits.getTopCount();
+        }
+
+        @Override
+        public final int hashCode() {
+            return HASH_CODE;
```
Comment (ScrapCodes) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/model/Settings.java]: Why are you overriding this ?

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/model/Settings.java [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+        }
+
+    }
+
+    public static class EsURLComboBoxModel {
```
Comment (ScrapCodes) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/model/Settings.java]: static is bad

File: plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/model/Settings.java (hunk 3 above)
Comment (keerath) [inline plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/model/Settings.java]: Checkstyle complains if you override equals without overriding the hash code function

**Pull request closed without merging**